nemo/collections/tts/modules/aligner.py

```diff
@@ -143,18 +143,21 @@ def get_mean_distance_for_word(l2_dists, durs, start_token, num_tokens):
 
         return dist_sum / total_frames
 
-    def forward(self, queries, keys, mask=None, attn_prior=None):
+    def forward(self, queries, keys, mask=None, attn_prior=None, conditioning=None):
         """Forward pass of the aligner encoder.
 
         Args:
             queries (torch.tensor): B x C x T1 tensor (probably going to be mel data).
             keys (torch.tensor): B x C2 x T2 tensor (text data).
             mask (torch.tensor): B x T2 x 1 tensor, binary mask for variable length entries (True = mask element, False = leave unchanged).
             attn_prior (torch.tensor): prior for attention matrix.
+            conditioning (torch.tensor): B x T2 x 1 conditioning embedding
         Output:
             attn (torch.tensor): B x 1 x T1 x T2 attention mask. Final dim T2 should sum to 1.
             attn_logprob (torch.tensor): B x 1 x T1 x T2 log-prob attention mask.
         """
+        if conditioning is not None:
+            keys = keys + conditioning.transpose(1,2)
```

Comment:
I think it would make more sense to condition/concatenate the speaker embedding to the spectrogram features, rather than the text features. Parsing spectrogram/audio benefits heavily from knowing the speakers vocal characteristics ahead of time.

(Text + speaker) is an ambiguous one to many mapping, while (spectrogram + speaker) to text is a deterministic many to one map. So conditioning the spectrogram parser will be more robust.

Or you could condition both, I guess.

Reply:
I guess we can do that (add condition vector to spectrogram also) but the reason why I did add to text vector only is because of dimensions. So dimensions of all vectors (text, speaker etc) are of same dimension but spectrogram vector dimension varies. So adding won't be that straight forward.